### PR TITLE
chore: remove `prettier` as a dependency

### PR DESCRIPTION
**Pull request by AdrianGonz97**
`prettier` is [ginormous](https://npmgraph.js.org/?q=svelte-add#select=exact%3Aprettier%403.3.2&sizing=) and is the largest dependency that users have to download every time they call any of our adders. 

Rather than having it as a direct dependency and using it's api in code, we can check if `prettier` is installed in their project and run the formatter via its CLI after all of the adders have executed and after dependency installations have occurred (covers the case of if `node_modules` isn't installed yet).

### CODE REVIEW

**Comment by benmccann** [comment]
sounds like a good idea to me. I don't think it should affect this, but I'd like to move the `prettier` setup (along with all the others) from `create-svelte` to this repo. Rich confirmed today that he was okay with that

**Comment by AdrianGonz97** [comment]
> sounds like a good idea to me. I don't think it should affect this, but I'd like to move the `prettier` setup (along with all the others) from `create-svelte` to this repo. Rich confirmed today that he was okay with that

cool! and yea, this PR wouldn't block the ability to add a prettier adder

**Review by manuel3108 — COMMENTED** [review]
I like the idea, but I think there are a couple of problems at the current point in time.

Let's start by saying that we won't be able to format the files if the user decides to skip installing dependencies. That might cause some files to have a weird indentation or formatting. Although as of know, all of the adders to behave not to badly.

If you are creating a new project through `svelte-add` right now, the only way you will get prettier installed and thus your files formatted would be if you select to create a new kit project and additionally select to install add prettier and decided to install the dependencies.  The projects created by `create-vite` do not seem to bring prettier by default. In total I think that's a lot of conditions that need to apply to make this work. Looking into the future at the other hand, where we will provide the prettier adder as part of the `svelte-add` cli, that makes a ton more sense.

And for the last thing I noticed I'm not sure if it belongs here. If I create a new kit project via the cli inside our monorepo (I usually use `./temp`), select to add prettier, and install the dependencies, the following is printed to my console:
```
◇  Failed to format files
│
■  [error] Cannot find package 'prettier-plugin-tailwindcss' imported from D:\dev\web\svelte-add\temp\noop.js
│
│
└  You're all set!
```

If I execute exactly the same setup in a different folder outside of our repo root (let's say `../svelte-add-temp`), then this error is not shown and everything works as expected.

---
To summarize I think this change makes a ton of sense dependency wise. Though at this point merging this might look like a step back, because our own `prettier` adder is not ready yet. I'll leave it to you guys to decide if we should merge this at this point, as I really don't have a strong opinion in any direction right now.

**Comment by AdrianGonz97** [comment]
> Let's start by saying that we won't be able to format the files if the user decides to skip installing dependencies. That might cause some files to have a weird indentation or formatting. Although as of know, all of the adders to behave not to badly.

I think having the adders format the modified files is more of a nice-to-have rather than a required feature. I don't think it's particularly worth it at the cost of doubling the total package size + dependencies for our adders.

Additionally, formatting for the users who _don't_ have `prettier` also has its own downsides as well. Most notably for those that are using different formatters (biomejs, eslint stylistic, etc). We're now prescribing a certain formatting style that likely doesn't conform to whichever a user has set and will probably have to be reformatted by the user anyway.

> And for the last thing I noticed I'm not sure if it belongs here. If I create a new kit project via the cli inside our monorepo (I usually use ./temp), select to add prettier, and install the dependencies, the following is printed to my console:

this looks like an edge-case because of monorepos, although I think I have an idea as to how to fix it

> Though at this point merging this might look like a step back, because our own prettier adder is not ready yet.

I do see your point with not having a prettier adder, though. I don't think it'll be that big of a lift to add it, so I can open a new PR to add it as a pre-req for this PR

**Comment by manuel3108** [comment]
> Additionally, formatting for the users who _don't_ have `prettier` also has its own downsides as well. Most notably for those that are using different formatters (biomejs, eslint stylistic, etc). We're now prescribing a certain formatting style that likely doesn't conform to whichever a user has set and will probably have to be reformatted by the user anyway.

Havn't looked at it from this side, that's a strong point.

> I do see your point with not having a prettier adder, though. I don't think it'll be that big of a lift to add it, so I can open a new PR to add it as a pre-req for this PR

Let's take our time, as there are some issues that we need to solve before properly being able to manage this, see #445, #411

**Review by manuel3108 — APPROVED** [review]
